element-template-generator/core/src/main/java/io/camunda/connector/generator/java/util/ConfigurationUtil.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH
+ * under one or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information regarding copyright
+ * ownership. Camunda licenses this file to you under the Apache License,
+ * Version 2.0; you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.camunda.connector.generator.java.util;
+
+import io.camunda.connector.generator.api.GeneratorConfiguration;
+import io.camunda.connector.generator.api.GeneratorConfiguration.ConnectorElementType;
+import io.camunda.connector.generator.java.annotation.ElementTemplate;
+import java.util.Arrays;
+import java.util.stream.Collectors;
+
+public class ConfigurationUtil {
+
+  public static GeneratorConfiguration fromAnnotation(
+      ElementTemplate annotation, GeneratorConfiguration override) {
+    var templateId = annotation.id();
+    if (override.templateId() != null) {
+      templateId = override.templateId();
+    }
+    var templateName = annotation.name();
+    if (override.templateName() != null) {
+      templateName = override.templateName();
+    }
```

Comment:
🙃nitpicking: Maybe rewrite all of these to a one liner:
`var templateName = Optional.ofNullable(override.templateName()).orElseGet(() -> annotation.name())`